src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/DockerRegistryManager.cs

```diff
@@ -63,4 +111,50 @@ public static void ShutdownDockerRegistry(ITestOutputHelper testOutput)
                 .Should().Pass();
         }
     }
+
+    private static void EnsureRegistryLoaded(string registryBaseUri, string? containerRegistryId, ILogger logger, ITestOutputHelper testOutput)
+    {
+        const int registryLoadMaxRetry = 10;
+        const int registryLoadTimeout = 1000; //ms
+
+        using HttpClient client = new();
```

Comment:
consider an HttpClient singleton cleaned up at the test pass' conclusion instead of disposing on each call.

* https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines
* https://medium.com/@nuno.caneco/c-httpclient-should-not-be-disposed-or-should-it-45d2a8f568bc

Reply:
test class fixture is already a singleton in test run context. Test class fixture is initialized only once.
This method will only be re-run in case unexpected failure and retry.